OrbitSsh/include/OrbitSsh/DirectTcpChannelManager.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_SSH_DIRECT_TCP_CHANNEL_MANAGER_H_
+#define ORBIT_SSH_DIRECT_TCP_CHANNEL_MANAGER_H_
+
+#include <libssh2.h>
+
+#include <memory>
+
+#include "Channel.h"
+#include "ResultType.h"
+
+namespace OrbitSsh {
+
+class DirectTcpChannelManager {
+ public:
+  enum State {
+    kNotInitialized,
+    kRunning,
+    kSentEOFToRemote,
+    kRemoteSentEOFBack,
+    kClosed
+  };
+
+  DirectTcpChannelManager(Session* session_ptr, int remote_port,
+                          std::string remote_host = "localhost");
```

Comment:
I wouldn't put a default here. It makes it even harder to guess what this number in remote_port might be. I would remove the default and reorder such that remote_host is before remote_port. That makes the API more familiar.

Reply:
This is a little tricky, I flip it, rename it and I am going to write a comment in the code for this. 

`remote_host` here might seem like it is the ip address/hostname of the remote ssh server that we want to connect to. At the point of this usage the connection to the remote ssh server is already established. This DirectTcp class connects to ab arbitrary ip address, via the remote host. In the most use cases (and certainly ours) this address is localhost.